<!DOCTYPE html>
<html lang="en">
<head>
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/katex.min.css" integrity="sha384-zB1R0rpPzHqg7Kpt0Aljp8JPLqbXI3bhnPWROx27a9N0Ll6ZP/+DiW/UqRcLbRjq" crossorigin="anonymous">

    <!-- The loading of KaTeX is deferred to speed up page rendering -->
    <script defer src="https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/katex.min.js" integrity="sha384-y23I5Q6l+B6vatafAwxRu/0oK/79VlbSz7Q9aiSZUvyWYIYsd+qj+o24G5ZU2zJz" crossorigin="anonymous"></script>

    <!-- To automatically render math in text elements, include the auto-render extension: -->
    <script defer src="https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/contrib/auto-render.min.js" integrity="sha384-kWPLUVMOks5AQFrykwIup5lo0m3iMkkHrD0uJ4H5cjeGihAutqP0yW0J6dpFiVkI" crossorigin="anonymous"
        onload="renderMathInElement(document.body);"></script>

  <meta charset="utf-8">
  <meta name="generator" content="pandoc">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
  <title>scVI</title>
  <link rel="stylesheet" href="style.css">
  <script src="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.11.1/katex.min.js"></script>
  <script>document.addEventListener("DOMContentLoaded", function () {
   var mathElements = document.getElementsByClassName("math");
   var macros = [];
   for (var i = 0; i < mathElements.length; i++) {
    var texText = mathElements[i].firstChild;
    if (mathElements[i].tagName == "SPAN") {
     katex.render(texText.data, mathElements[i], {
      displayMode: mathElements[i].classList.contains('display'),
      throwOnError: false,
      macros: macros,
      fleqn: false
     });
  }}});
  </script>
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.11.1/katex.min.css" />
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<article>
<header>
<div id="chapter">1</div>
<h1 class="title">scVI</h1>
</header>



<div id="collapsiblemenu">
  <button class="collapsible">
    <div class="shortthickbar"></div>
    <div class="shortthickbar"></div>
    <div class="shortthickbar"></div>
  </button>
  <div class="content">
    <ul>
    <li><a href="index.html">Home</a></li>
    </ul>
    <ul>
    <li><a href="#model">Model</a></li>
    <li><a href="#inference">Inference</a></li>
    <li><a href="#training">Training</a></li>
    <li><a href="#tasks">Tasks</a></li>
    <li><a href="#math-to-code">Math to code</a></li>
    <li><a href="#bibliography">References</a></li>
    </ul>
  </div>
</div>


<section id="model" class="level1">
<h1>Model</h1>
<p>The output of a scRNAseq experiment is a matrix of counts with&nbsp;<span class="math inline">N</span> rows (the number of cells) and&nbsp;<span class="math inline">G</span> columns (the number of genes), where each entry&nbsp;<span class="math inline">x_{ng}</span> is an integer representing how many transcripts of gene&nbsp;<span class="math inline">g</span> where seen in cell&nbsp;<span class="math inline">n</span>. scVI is a generative hierarchical Bayesian model for scRNAseq data with conditional distributions parametrized by neural networks for each gene.<span class="citation" data-cites="scvi"><span><label for="sn-0" class="margin-toggle sidenote-number"></label><input type="checkbox" id="sn-0" class="margin-toggle" /><span class="sidenote">Lopez et al., <span>“Deep Generative Modeling for Single-Cell Transcriptomics,”</span> <em>Nature Methods</em> 15, no. 12 (2018): 1053–58.</span></span></span> There are technical variables to account for different batches (<span class="math inline">s_n</span>) and for library size (<span class="math inline">l_n</span>, which can be interpreted as cell size or sequencing depth). Thus the number of networks being trained is&nbsp;<span class="math inline">2 \cdot G \cdot K</span>, where&nbsp;<span class="math inline">K</span> is the total the number of batches (datasets).</p>
<p>Conditional distribution&nbsp;<span class="math inline">p\left(x_{n g} \mid z_{n}, l_n, s_n \right)</span> is a zero-inflated negative binomial distribution (ZINB) to model the kinetics of stochastic gene expression with some entries replaced by zeros. It can also be modelled using Negative binomial or Zero-inflated negative binomial using the <a href="https://docs.scvi-tools.org/en/stable/_modules/scvi/model/_scvi.html"><code>gene_likelihood</code></a> argument.</p>
<p>The neural networks&nbsp;<span class="math inline">f^g_{w}</span> and&nbsp;<span class="math inline">f^g_{h}</span> use dropout regularization and batch nomalization to model gene expression while accounting for library sizes and batch effects respectively. Each network typically has 3 fully connected-layers, with 128-256 nodes each. The activation functions are ReLU, exponential, or linear.&nbsp;<span class="math inline">f_{w}</span> has a final softmax layer to represent normalized expected frequencies of gene expression as in. Weights for some layers are shared between&nbsp;<span class="math inline">f_{w}</span> and&nbsp;<span class="math inline">f_{h}</span>.</p>
<figure>
<img src="assets/scvi_annotated_graphical_model.png" alt="Fancy plot" /><figcaption aria-hidden="true">Fancy plot</figcaption>
</figure>
</section>
<section id="inference" class="level1">
<h1>Inference</h1>
<p>Detail the inference objective</p>
</section>
<section id="training" class="level1">
<h1>Training</h1>
<p>Any details that aren’t clear in manuscripts but are important for training.</p>
</section>
<section id="tasks" class="level1">
<h1>Tasks</h1>
<p>Here we put the mathematical description of tasks.</p>
</section>
<section id="math-to-code" class="level1">
<h1>Math to code</h1>
<p>Table for each variable, what it’s variable name is in the code</p>
</section>
<section id="bibliography" class="level1 unnumbered">
<h1 class="unnumbered">References</h1>
<div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-scvi" class="csl-entry" role="doc-biblioentry">
Lopez, Romain, Jeffrey Regier, Michael B Cole, Michael I Jordan, and Nir Yosef. <span>“Deep Generative Modeling for Single-Cell Transcriptomics.”</span> <em>Nature Methods</em> 15, no. 12 (2018): 1053–58.
</div>
</div>
</section>

<div id="lastupdate">
Last updated: Mon May 17 22:03:31 UTC 2021
</div>
</article>


<script>
var coll = document.getElementsByClassName("collapsible");
var i;

for (i = 0; i < coll.length; i++) {
  coll[i].addEventListener("click", function() {
    this.classList.toggle("active");
    var content = this.nextElementSibling;
    if (content.style.display === "block") {
      content.style.display = "none";
    } else {
      content.style.display = "block";
    }
  });
}
</script>

</body>
</html>
